src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
+): GroupsMap {
+  const result: GroupsMap = {};
```

Comment:
The `result` variable should also be typed as `GroupsMap<T>` for consistency and to leverage the generic type.